src/Components/Blazor/Blazor/src/Rendering/WebAssemblyRenderer.cs

```diff
@@ -110,5 +114,79 @@ protected override void HandleException(Exception exception)
                 Console.Error.WriteLine(exception);
             }
         }
+
+        /// <inheritdoc />
+        public override Task DispatchEventAsync(int eventHandlerId, UIEventArgs eventArgs)
+        {
+            // Be sure we only run one event handler at once. Although they couldn't run
+            // simultaneously anyway (there's only one thread), they could run nested on
+            // the stack if somehow one event handler triggers another event synchronously.
+            // We need event handlers not to overlap because (a) that's consistent with
+            // server-side Blazor which uses a sync context, and (b) the rendering logic
+            // relies completely on the idea that within a given scope it's only building
+            // or processing one batch at a time.
+            //
+            // The only currently known case where this makes a difference is in the E2E
+            // tests in ReorderingFocusComponent, where we hit what seems like a Chrome bug
+            // where mutating the DOM cause an element's "change" to fire while its "input"
+            // handler is still running (i.e., nested on the stack) -- this doesn't happen
+            // in Firefox. Possibly a future version of Chrome may fix this, but even then,
+            // it's conceivable that DOM mutation events could trigger this too.
+
+            if (isDispatchingEvent)
+            {
+                var info = new IncomingEventInfo(eventHandlerId, eventArgs);
+                deferredIncomingEvents.Enqueue(info);
+                return info.TaskCompletionSource.Task;
+            }
+            else
+            {
+                try
+                {
+                    isDispatchingEvent = true;
+                    return base.DispatchEventAsync(eventHandlerId, eventArgs);
+                }
+                finally
+                {
+                    isDispatchingEvent = false;
+
+                    if (deferredIncomingEvents.Count > 0)
+                    {
+                        ProcessNextDeferredEvent();
+                    }
+                }
+            }
+        }
+
+        private void ProcessNextDeferredEvent()
+        {
+            var info = deferredIncomingEvents.Dequeue();
+            var task = DispatchEventAsync(info.EventHandlerId, info.EventArgs);
+            task.ContinueWith(_ =>
```

Comment:
Is there a really good reason to use `ContinueWith` here instead of `await`. I know there's no scheduler in this use case, but it's seemed like we've had bugs in most of our usage of `ContinueWith` so I like to avoid it.

Reply:
I was just avoiding `async void` since there's no drawback to the `ContinueWith` here. However it doesn't really matter, so based on your preference I've changed this now.

Reply:
You can also `async Task` and then invoke it as fire-and-forget.

Reply:
I think that would be more misleading. It suggests the upstream code is responsible for error handling.